### PR TITLE
Export All Grades to CSV (move to new code base)

### PR DESCRIPTION
**Pull request by jbarthelmess**
Creates a CSV Report with all gradeables and their final scores.
- Top row is the setup, each gradeable name in the top row also has it's respective max score.
- Student's ordered by registration section (how gradeables were returned)

### CODE REVIEW

File: site/app/controllers/admin/ReportController.php [hunk 1]
```diff
@@ -37,8 +37,52 @@ public function showReportPage() {
...
+            }
+            $csv_output .= implode(",",$student_line).$nl;
+        }
         
+        header("Content-Type: text/plain");
+        header('Content-Disposition: attachment; filename=hwserver-report.csv');
+        header("Content-Length: " . strlen($csv_output));
+        echo $csv_output;
+        exit;
+        //$this->core->getOutput()->renderOutput(array('admin', 'Report'), 'showReportUpdates');
```
Comment (MasterOdin) [inline site/app/controllers/admin/ReportController.php]: get rid of this?

File: site/app/controllers/admin/ReportController.php [hunk 2]
```diff
@@ -37,8 +37,52 @@ public function showReportPage() {
...
+                $student_line[] = $student[$grade_id];
+            }
+            $csv_output .= implode(",",$student_line).$nl;
+        }
         
+        header("Content-Type: text/plain");
+        header('Content-Disposition: attachment; filename=hwserver-report.csv');
+        header("Content-Length: " . strlen($csv_output));
+        echo $csv_output;
+        exit;
```
Comment (jbarthelmess) [inline site/app/controllers/admin/ReportController.php]: Not sure what you mean when you say you want me to return <code>$csv_output</code>.  Also <code>displayOutput()</code> was never getting called (we were printing by <code>echo($core->getOutput()->getOutput())</code> and I changed that.

Comment (MasterOdin) [inline site/app/controllers/admin/ReportController.php]: Eh, ignore the `$csv_output` return for now and that can be done when we actually add tests for this function.

Change [this line](https://github.com/Submitty/Submitty/blob/master/site/public/index.php#L212) to be instead `$core->getOutput()->displayOutput()`. Then add what I mentioned above within the Output library.

Comment (jbarthelmess) [inline site/app/controllers/admin/ReportController.php]: ...../ )
.....' /
---' (_____
......... ((__)
..... _ ((___)
....... -'((__)
--.___((_)

File: site/app/controllers/admin/ReportController.php [hunk 3]
```diff
@@ -37,8 +37,48 @@ public function showReportPage() {
...
+            $g_id = $gradeable->getId();
+            if(!isset($results['header_model'][$g_id])) {
+                $results['header_model'][$g_id] = $g_id.": ".($gradeable->getTotalAutograderNonExtraCreditPoints() + $gradeable->getTotalTANonExtraCreditPoints());
+            }
+            $results[$student_id][$g_id] = $gradeable->getGradedTAPoints() + $gradeable->getGradedAutograderPoints();
+        }
         
+        $nl = "\n";
+        $csv_output = "";
+        $filename = $_REQUEST['course']."CSVReport.csv";
```
Comment (MasterOdin) [inline site/app/controllers/admin/ReportController.php]: use `$this->core->getConfig()->getCourse()`

**Pull request merged**